tcms/search/advanced_search.py

```diff
@@ -33,21 +33,8 @@ def advance_search(request):
     run_form.populate(data)
     all_forms = (plan_form, case_form, run_form)
 
-    errors = []
-    for form in all_forms:
-        if form.is_valid():
-            errors.append(form.errors)
-
-    if errors or not data:
-        product_choice = []
-        for product in Product.objects.all():
-            product_choice.append((product.pk, product.name))
-        plan_type_choices = PlanType.objects.all()  # pylint: disable=unused-variable
-        errors = _fmt_errors(errors)
-        priorities = Priority.objects.filter(  # pylint: disable=unused-variable
-            is_active=True).order_by('value')
-
-        return render(request, 'search/advanced_search.html', locals())
+    if _render_errors(request, data, all_forms):
+        return _render_errors(request, data, all_forms)
```

Comment:
This statement has a side effect. You call `_render_errors()` once to evaluate the conditional expression on line 36. When this is True you call the function a second time to return it's result (which is the same both times).

Assign the result of this function to a variable (hint: render() returns a string) and if the string isn't empty return its value.